x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectRealm.java

```diff
@@ -178,8 +184,21 @@ private void buildUserFromClaims(JWTClaimsSet claims, ActionListener<Authenticat
             return;
         }
 
+        final Map<String, Object> tokenMetadata = new HashMap<>();
+        tokenMetadata.put("id_token_hint", claims.getClaim("id_token_hint"));
+        tokenMetadata.put("oidc_realm", this.name());
```

Comment:
I think this is redundant since we have the realm name in the `Authentication` which is serialized together with this metadata.